src/components/calendarpage.jsx

```diff
@@ -131,13 +139,34 @@ export function CalendarPage() {
     periodTracker.createPeriodEntry(periodEntryData);
   }
 
+  const DayCell = ({ info }) => {
+    // Check if date overlaps with a period
+    for (let i = 0; i < periods.length; i++) {
+      const period = periods[i];
+      const dayCellDate = info.date;
+      const startDate = stringToDate(period.startDate);
+      const endDate = stringToDate(period.endDate);
+      if (dayCellDate >= startDate && dayCellDate <= endDate) {
+        return (<div></div>);
+      }
+    }
```

Comment:
```suggestion
     const dayCellDate = info.date;
     const isOverlap = periods.some(period => {
        const startDate = stringToDate(period.startDate);
        const endDate = stringToDate(period.endDate);
        return dayCellDate >= startDate && dayCellDate <= endDate;
      })
      if (isOverlap) {  
        return (<div></div>);
      }
    }
```

Reply:
also might be good to have a function isOverlap

```
function isInPeriod(aDate, aPeriod) {
        const startDate = stringToDate(period.startDate);
        const endDate = stringToDate(period.endDate);
        return aDate >= startDate && aDate <= endDate;
}

and then..
if (periods.some(period => isInPeriod(info.date, period))) {
  return (<div></div>);
}
```

The isOverlap would go really nicely on a Period object :)

Reply:
Getting an error that `.some` is not a function on `periods`
This is a hack week project, so I'm not super dedicated to improving the code if it works. but feel free to submit these improvements in a separate PR

Reply:
Yup, mostly wanted to highlight that using builtin collection functions can help get rid of the `for loop` mechanics and draw more attention to what the for loop is doing :)